easybuild/easyblocks/generic/systemcompiler.py

```diff
@@ -115,9 +115,29 @@ def extra_options():
         return extra_vars
 
     def __init__(self, *args, **kwargs):
-        """Extra initialization: determine system compiler version and prefix."""
+        """Extra initialization: keep track of values that may change due to modifications to the version."""
         super(SystemCompiler, self).__init__(*args, **kwargs)
 
+        # Keep track of original values of vars that are subject to change, for restoring later.
+        # The version is determined/matched from the installation and the installdir is determined from the system (the original is used
```

Comment:
line too long, fixed in https://github.com/ocaisa/easybuild-easyblocks/pull/13, should then be good to go in